extension/src/plots/model/collect.ts

```diff
@@ -413,6 +426,46 @@ export const collectData = (
   return acc
 }
 
+const collectFlexiblePlot = (
+  acc: Record<string, boolean>,
+  path: string,
+  plot: TemplatePlot
+  // eslint-disable-next-line sonarjs/cognitive-complexity
+) => {
+  acc[path] = false
+  const files = new Set()
+  for (const value of Object.values(plot.datapoints || {}).flat()) {
+    const filename = (
+      value?.dvc_data_version_info as {
+        filename: string | undefined
+      }
+    )?.filename
+    if (filename) {
+      files.add(filename)
+    }
+    if (files.size > 1) {
+      acc[path] = true
+      return
+    }
+  }
+}
```

Comment:
Both here and in `getSpecEncodingUpdate` we need to remember that that filename is not the only field differentiating "normal" plot from "flexible". We need to count "field" occurrences too - one might make a plot out of two columns from single `csv` file for example. Take a look at `isMultiSourcePlot` from 
https://github.com/iterative/vscode-dvc/pull/2379.

Reply:
Thank you, this was just garbage to be able to show something.

Reply:
Sure thing. I guess that's where marking particular points in different style might come in handy